public/app/core/components/RolePicker/TeamRolePicker.tsx

```diff
@@ -5,35 +5,60 @@ import { contextSrv } from 'app/core/core';
 import { Role, AccessControlAction } from 'app/types';
 
 import { RolePicker } from './RolePicker';
-// @ts-ignore
 import { fetchTeamRoles, updateTeamRoles } from './api';
 
 export interface Props {
   teamId: number;
   orgId?: number;
   roleOptions: Role[];
   disabled?: boolean;
-  builtinRolesDisabled?: boolean;
+  onApplyRoles?: (newRoles: Role[]) => void;
+  pendingRoles?: Role[];
+  /**
+   * Set whether the component should send a request with the new roles to the
+   * backend in TeamRolePicker.onRolesChange (apply=false), or call {@link onApplyRoles}
+   * with the updated list of roles (apply=true).
+   *
+   * Besides it sets the RolePickerMenu's Button title to
+   *   * `Update` in case apply equals false
+   *   * `Apply` in case apply equals true
+   *
+   * @default false
+   */
+  apply?: boolean;
 }
 
-export const TeamRolePicker: FC<Props> = ({ teamId, orgId, roleOptions, disabled, builtinRolesDisabled }) => {
+export const TeamRolePicker: FC<Props> = ({
+  teamId,
+  roleOptions,
+  disabled,
+  onApplyRoles,
+  pendingRoles,
+  apply = false,
+}) => {
   const [{ loading, value: appliedRoles = [] }, getTeamRoles] = useAsyncFn(async () => {
     try {
-      return await fetchTeamRoles(teamId, orgId);
+      if (apply && pendingRoles?.length! > 0) {
```

Comment:
```suggestion
      if (apply && Boolean(pendingRoles?.length)) {
```
`pendingRoles` can be `undefined` and in that case you're comparing `undefined` with a number, which is what TS complains about.

Reply:
Fixed.